src/cdk/schema.rs

```diff
@@ -0,0 +1,580 @@
+    fn from(map: HashMap<String, V, Hasher>) -> Self {
+        Self::HashMap(map)
+    }
+}
+
+pub trait PropertyBag {
+    // Retrieves the property with the provided CloudFormation name, if any.
+    fn property(&self, name: &str) -> Option<Property>;
+}
+
+// Information about an AWS CDK construct class.
+#[derive(Clone, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct CfnResource {
+    // The type name of the construct class.
+    pub construct: TypeName,
+
+    // The properties declared by the construct class, indexed by their
+    // CloudFormation name (e.g: `"BucketName"`).
+    pub(super) properties: Map<Property>,
+
+    // The attributes declared by the construct class, indexed by their
+    // CloudFormation name (e.g: `"Arn"`).
+    pub(super) attributes: Map<Property>,
+}
+
+impl CfnResource {
+    // Creates a new `ConstructInfo` with the provided information.
+    #[cfg(test)]
+    pub const fn new(
+        construct: TypeName,
+        properties: Map<Property>,
+        attributes: Map<Property>,
+    ) -> Self {
+        Self {
+            construct,
+            properties,
+            attributes,
+        }
+    }
+
+    // Retrieves the attribute with the provided CloudFormation name, if any.
+    pub fn attribute(&self, name: &str) -> Option<&Property> {
+        self.attributes.get(name)
+    }
+}
+
+impl PropertyBag for CfnResource {
+    fn property(&self, name: &str) -> Option<Property> {
+        self.properties.get(name).map(Clone::clone)
+    }
+}
+
+// Information about an AWS CDK struct.
+#[derive(Clone, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct DataType {
+    /**
+     * The name of the type.
+     */
+    pub name: TypeName,
+
+    // The properties declared by the AWS CDK struct, indexed by their
+    // CloudFormation name (e.g: `"BucketName"`).
+    #[serde(bound(deserialize = "Map<Property>: serde::Deserialize<'de>"))]
+    pub(super) properties: Map<Property>,
+}
+
+impl DataType {
+    // Creates a new `StructInfo` with the provided properties.
+    #[cfg(test)]
+    pub const fn new(name: TypeName, properties: Map<Property>) -> Self {
+        Self { name, properties }
+    }
+}
+
+impl PropertyBag for DataType {
+    fn property(&self, name: &str) -> Option<Property> {
+        self.properties.get(name).map(Clone::clone)
+    }
+}
+
+// A multi-language aware type name.
+#[derive(serde::Deserialize)]
+pub struct TypeName {
+    // The name of the type in TypeScript.
+    #[cfg(feature = "typescript")]
+    pub typescript: TypeScriptName,
+
+    // The name of the type in .NET.
+    #[cfg(feature = "csharp")]
+    pub csharp: DotNetName,
+
+    // The name of the type in Go.
+    #[cfg(feature = "golang")]
+    pub golang: GolangName,
+
+    // The name of the type in Go.
+    #[cfg(feature = "java")]
+    pub java: JavaName,
+
+    // The name of the type in Python.
+    #[cfg(feature = "python")]
+    pub python: PythonName,
+}
+
+impl Clone for TypeName {
```

Comment:
I'm 99% sure since these are all just unconditional clones you can instead just put a `#[derive(Clone)]` on the 
`pub struct TypeName {` above.